### PR TITLE
Adding information about RISC Zero Secure SDLC

### PR DESCRIPTION
**Pull request by kevinnassery**
Adding secure-sdlc information.

### CODE REVIEW

**Review by pdg744 — CHANGES_REQUESTED** [review]
Thanks Kevin! 

Can we add some links to make sure users can find related assets? In particular, users should be able to find completed audit reports from this page. BSIMM report is mb also worth including.

File: website/api/secure-sdlc.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# RISC Zero Secure Development Life-cycle
+
+RISC Zero believes in a comprehensive risk-based approach to security throughout our product and software development life-cycle.
+
+This page provides an outline of the key practices that we utilize to harden our products and services against threats. These security techniques are continuously evolving as we learn from our experiences and adapt to new threats, and we encourage you to provide feedback or suggestions for improvement.
+
+The key mechanisms that drive our process evolution are:
+* External critique from customers, partners, and the security community
+* Escape analysis of production defects and postmortem analysis of security incidents
+* Identifying new potential activities based on industry frameworks including [NIST SSDF](https://csrc.nist.gov/Projects/ssdf), [ISO 27002:2022](https://www.iso.org/standard/75652.html), [BSIMM 14](https://www.synopsys.com/software-integrity/software-security-services/bsimm-maturity-model.html), [the ReKT test](https://www.risczero.com/blog/risc-zero-take-the-rekt-test), and [OWASP SAAAM](https://owasp.org/www-project-samm/).
```
Comment (pdg744) [inline website/api/secure-sdlc.md:10]: Our link convention is to use reference links in-line and include anchors at the bottom of the page -- can you move the links to the bottom? 

```suggestion
* Identifying new potential activities based on industry frameworks including [NIST SSDF], [ISO 27002:2022], [BSIMM 14], [the ReKT test], and [OWASP SAAAM].
```

File: website/api/secure-sdlc.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+* Use of a maturity framework (known as FACTORS) to self-assess process maturity in the following areas:
```
Comment (pdg744) [inline website/api/secure-sdlc.md:11]: I'm not finding a reference for `FACTORS` after a quick google search. Can we add a link? 
```suggestion
* Use of a maturity framework (known as [FACTORS]) to self-assess process maturity in the following areas:
```

File: website/api/secure-sdlc.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+* The use of secure-by-design components, libraries and frameworks.
+* The use of languages that provide memory safety advantages.
+* The use of static analysis tools to identify common security vulnerabilities.
+* The use of software composition analysis tools to identify potential vulnerabilities in third-party libraries.
+* The use of protected branches and a peer review process that's backed by named code owners who can provide substantive review including security best practices.
+
+## 4. Verify
+The verification phase is where the security of the product is tested. This phase includes a combination of manual and automated testing, including:
+
+* External deep dive audits by leading security firms.
```
Comment (pdg744) [inline website/api/secure-sdlc.md:58]: Can we include link to audits and mentions of teams doing auditing here?

File: website/api/secure-sdlc.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+* Targeted fuzzing and dynamic security testing.
+* Opportunistic use of formal verification in critical components of the system.
```
Comment (pdg744) [inline website/api/secure-sdlc.md:60]: Mb we want to link to this video? 
https://www.youtube.com/watch?v=tQ1TQBIk_Lc

File: website/api/secure-sdlc.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+* The development and execution of security invariants and associated test cases.
+
+## 5. Release
+This phase validates if the objectives of the security plan have been met. This minimally includes verifying that:
+* Findings from phase 4 have been addressed.
+* Any necessary models to the formal threat model have been made.
```
Comment (pdg744) [inline website/api/secure-sdlc.md:66]:
```suggestion
* Any necessary changes to the formal threat model have been made.
```

File: website/api/secure-sdlc.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+* Operational security processes are in place (e.g. IR and Bug bounty).
```
Comment (pdg744) [inline website/api/secure-sdlc.md:67]:
```suggestion
* Operational security processes are in place (e.g. IR and [Bug bounty]).
```

File: website/api/secure-sdlc.md (hunk 4 above)
Comment (kevinnassery) [reply website/api/secure-sdlc.md:60]: We will have a deeper story on verification soonish so let's make an update when that work is completed.

File: website/api/secure-sdlc.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+RISC Zero believes in a comprehensive risk-based approach to security throughout our product and software development life-cycle.
+
+This page provides an outline of the key practices that we utilize to harden our products and services against threats. These security techniques are continuously evolving as we learn from our experiences and adapt to new threats, and we encourage you to provide feedback or suggestions for improvement.
+
+The key mechanisms that drive our process evolution are:
+* External critique from customers, partners, and the security community
+* Escape analysis of production defects and postmortem analysis of security incidents
+* Identifying new potential activities based on industry frameworks including [NIST SSDF], [ISO 27002:2022], [BSIMM 14], [the ReKT test], and [OWASP SAAAM].
+* Use of a maturity framework (known as [FACTORS]) to self-assess process maturity in the following areas:
```
Comment (pdg744) [inline website/api/secure-sdlc.md:11]: [FACTORS] needs a linik

Comment (kevinnassery) [reply website/api/secure-sdlc.md:11]: Synopsys had this IP at one point, but it's faded away a bit.  I think in the future we can maybe blog about it then link back to that..

File: website/api/secure-sdlc.md (hunk 3 above)
Comment (kevinnassery) [reply website/api/secure-sdlc.md]: So, I think the auditing link is sufficient so we don't duplicate where we're tracking that

File: website/api/secure-sdlc.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+* [External deep dive audits] by leading security firms.
+* Targeted fuzzing and dynamic security testing.
+* Opportunistic use of formal verification in critical components of the system.
+* The development and execution of security invariants and associated test cases.
+
+## 5. Release
+This phase validates if the objectives of the security plan have been met. This minimally includes verifying that:
+* Findings from phase 4 have been addressed.
+* Any necessary changes to the formal threat model have been made.
+* Operational security processes are in place (e.g. IR and [Bug bounty]).
```
[the remark this answers is not in the document]
Comment (kevinnassery) [reply website/api/secure-sdlc.md:67]: done

**Comment by pdg744** [comment]
OK great -- links are working & tests are passing. 

Tagging @flaub and @jbruestle for review. 
You can see the content in context here: https://docs-website-e3i2vfo3u-risczero.vercel.app/api/next/secure-sdlc

I'll duplicate the changes to `1.0` once we finish review here.

**Review by pdg744 — APPROVED** [review]

**Pull request merged**